mula/scheduler/connectors/services/katalogus.py

```diff
@@ -22,69 +22,71 @@ def __init__(
-        self.organisations_new_boefjes_cache: dict = {}
+        self.new_boefjes_cache_lock = threading.Lock()
+        self.new_boefjes_cache: dict = {}
 
         # Initialise the cache.
         self.flush_caches()
 
     def flush_caches(self) -> None:
-        self.flush_organisations_plugin_cache()
-        self.flush_organisations_normalizer_type_cache()
-        self.flush_organisations_boefje_type_cache()
+        self.flush_plugin_cache()
+        self.flush_normalizer_cache()
+        self.flush_boefje_cache()
 
-    def flush_organisations_plugin_cache(self) -> None:
+    def flush_plugin_cache(self) -> None:
         self.logger.debug("Flushing the katalogus plugin cache for organisations")
 
-        with self.lock:
+        with self.plugin_cache_lock:
             # First, we reset the cache, to make sure we won't get any ExpiredError
-            self.organisations_plugin_cache.expiration_enabled = False
-            self.organisations_plugin_cache.reset()
+            self.plugin_cache.expiration_enabled = False
+            self.plugin_cache.reset()
 
             orgs = self.get_organisations()
             for org in orgs:
-                if org.id not in self.organisations_plugin_cache:
-                    self.organisations_plugin_cache[org.id] = {}
-                    self.organisations_new_boefjes_cache[org.id] = {}
+                if org.id not in self.plugin_cache:
+                    self.plugin_cache[org.id] = {}
 
                 plugins = self.get_plugins_by_organisation(org.id)
-                self.organisations_plugin_cache[org.id] = {plugin.id: plugin for plugin in plugins if plugin.enabled}
+                self.plugin_cache[org.id] = {plugin.id: plugin for plugin in plugins if plugin.enabled}
 
-            self.organisations_plugin_cache.expiration_enabled = True
+            self.plugin_cache.expiration_enabled = True
 
         self.logger.debug("Flushed the katalogus plugin cache for organisations")
 
-    def flush_organisations_boefje_type_cache(self) -> None:
+    def flush_boefje_cache(self) -> None:
         """boefje.consumes -> plugin type boefje"""
         self.logger.debug("Flushing the katalogus boefje type cache for organisations")
 
-        with self.lock:
+        with self.boefje_cache_lock:
             # First, we reset the cache, to make sure we won't get any ExpiredError
-            self.organisations_boefje_type_cache.expiration_enabled = False
-            self.organisations_boefje_type_cache.reset()
+            self.boefje_cache.expiration_enabled = False
+            self.boefje_cache.reset()
 
             orgs = self.get_organisations()
             for org in orgs:
-                self.organisations_boefje_type_cache[org.id] = {}
+                self.boefje_cache[org.id] = {}
 
-                for plugin in self.get_plugins_by_organisation(org.id):
+                plugins = self.get_plugins_by_organisation(org.id)
+                for plugin in plugins:
```

Comment:
Since you're not using the `plugins` variable, you might as well leave it out (like the previous implementation)